### ISSUE TITLE
registerHandlerClass function is not there?

### ISSUE THREAD

**Issue by borek2** [button/DeepLinkKit#111]
I have been upgrading to swift 3 for the passed hours i have put everything to work, except for DeepLinkKit. I am not able to find the function `registerHandlerClass` anymore. The weird thing is it is in the `DPLDeepLinkRouter.h` but when i command click on DeepLinkKit within my own code i get the following:

![dlk](https://cloud.githubusercontent.com/assets/2778466/18589505/5a3d2e32-7c2b-11e6-9fd3-506e065532b3.png)

I am not sure that this is a problem I have or this is a problem in the pod. I have deleted the pods, deleted Podfile.lock, deleted derived data, cleaned my project, everything, but it still doesn't show up. Does anyone know what to do?

**Comment by chrismaddern** [comment]
@borek2 I'm assuming that the header is in swift syntax because it was run through a Swift conversion? It looks like that converter made an error on interpreting the `registerHandlerClass:` method... although I'm not entirely sure why. 

How is DLK being imported? With an umbrella header import you should be able to leave this as Objective C syntax, no?

**Comment by borek2** [comment]
Thanks for the quick reply @chrismaddern !

I have added DeepLinkKit as a static library now, and it still gives problems. The actual function is there now, but i am not able to get a reference to it. With some trial and error (I am not very skillful in Objective-C) i noticed the following:

This is the line that gives problems:

`- (void)registerHandlerClass:(Class <DPLRouteHandler>)handlerClass forRoute:(NSString *)route;` 

When i use another class in instead of `DPLRouteHandler` it works fine (as, i can reference to it). But here is when it will cause problems. If you use a protocol (i created a new one to test this), it will fail to make a reference.

Sorry if it sounds a little vague, i only program in Swift so Objective C is sometimes a little vague to me. Might this problem have something to do with xCode 8?

**Comment by borek2** [comment]
Again it might be my lack of knowledge of Objective C, though in my understanding within the code DPLRouteHandler is both a protocol and a class? It is defined as a protocol in the file `DPLDeepLinkRouter.h`. But it has an implementation in `DPLRouteHandler.m`. Is it a naming problem maybe?

**Comment by doohoo** [comment]
+1, In the same seat as @borek2.

**Comment by chrismaddern** [comment]
@doohoo @borek2 It looks likely that there may have to be an interface change and bump to `2.0.0` for support for Xcode 8 / Swift 3. 

We'll take a look in the early week and try to get something out asap!

In the meantime, if either of you find a sensible interface change that works, feel free to fork / share.

**Comment by borek2** [comment]
@chrismaddern Will look into it! Thanks for the help!

**Comment by aborren** [comment]
I'm in the same situation. Hope you're able to sort it out asap 👍

**Comment by pantuspavel** [comment]
Hi @borek2 , thanks for reporting the issue.
[This branch](https://github.com/button/DeepLinkKit/tree/pavel/111_registerHandlerClass_swift3_interface) should fix the problem.
You can try it like this:
```
pod 'DeepLinkKit', :git => 'git@github.com:button/DeepLinkKit.git', :branch => 'pavel/111_registerHandlerClass_swift3_interface'
```

**Comment by borek2** [comment]
Awesome! Thanks so much! I will be testing this tomorrow @pantuspavel

**Comment by aborren** [comment]
It seems to work 👍  Thanks @pantuspavel

**Comment by doohoo** [comment]
Anyone got this working using Pods?

**Comment by pantuspavel** [comment]
@doohoo Hi!
PR in question was merged into master.
Please try:
```
pod 'DeepLinkKit', :git => 'git@github.com:button/DeepLinkKit.git', :branch => 'master'
```

**Comment by pantuspavel** [comment]
We restored fix branch, so you can try:
```
pod 'DeepLinkKit', :git => 'git@github.com:button/DeepLinkKit.git', :branch => 'pavel/111_registerHandlerClass_swift3_interface'
```

**Comment by doohoo** [comment]
@pantuspavel it didn't go to well.

Basically get:

```
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by pantuspavel** [comment]
@doohoo , seems like server rejects your connection.
Have you gone thorough this [guide](https://help.github.com/articles/error-permission-denied-publickey/)?

**Comment by doohoo** [comment]
Hehe, something is messup. Will check it out 👀 . Thanks.

**Comment by doohoo** [comment]
@pantuspavel just went for the `https` way instead :)

**Comment by pantuspavel** [comment]
No problem!
Glad it worked out for you.

**Comment by wessmith** [comment]
Sorry this took so long. The change is available now in [1.2.2](https://github.com/button/DeepLinkKit/releases/tag/1.2.2)
